src/metalperformanceshaders.cs

```diff
@@ -1135,5 +1525,1898 @@ interface MPSMatrixMultiplication {
+		IMPSNnPadding PaddingPolicy { get; set; }
+
+		[NullAllowed, Export ("label")]
+		string Label { get; set; }
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSNnFilterNode), Name = "MPSCNNConvolutionNode")]
+	interface MPSCnnConvolutionNode
+	{
+		[Static]
+		[Export ("nodeWithSource:weights:")]
+		MPSCnnConvolutionNode FromSource (MPSNnImageNode sourceNode, IMPSCnnConvolutionDataSource weights);
+
+		[Export ("initWithSource:weights:")]
+		IntPtr Constructor (MPSNnImageNode sourceNode, IMPSCnnConvolutionDataSource weights);
+
+		[NullAllowed, Export ("convolutionState")]
+		MPSCnnConvolutionStateNode ConvolutionState { get; }
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (MPSCnnConvolutionNode), Name = "MPSCNNFullyConnectedNode")]
+	interface MPSCnnFullyConnectedNode
+	{
+		[Static]
+		[Export ("nodeWithSource:weights:")]
+		MPSCnnFullyConnectedNode FromSource (MPSNnImageNode sourceNode, IMPSCnnConvolutionDataSource weights);
+
+		[Export ("initWithSource:weights:")]
+		IntPtr Constructor (MPSNnImageNode sourceNode, IMPSCnnConvolutionDataSource weights);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (MPSCnnConvolutionNode), Name = "MPSCNNBinaryConvolutionNode")]
+	interface MPSCnnBinaryConvolutionNode
+	{
+		[Static]
+		[Export ("nodeWithSource:weights:scaleValue:type:flags:")]
+		MPSCnnBinaryConvolutionNode FromSource (MPSNnImageNode sourceNode, IMPSCnnConvolutionDataSource weights, float scaleValue,MPSCnnBinaryConvolutionType type,MPSCnnBinaryConvolutionFlags flags);
+
+		[Export ("initWithSource:weights:scaleValue:type:flags:")]
+		IntPtr Constructor (MPSNnImageNode sourceNode, IMPSCnnConvolutionDataSource weights, float scaleValue,MPSCnnBinaryConvolutionType type,MPSCnnBinaryConvolutionFlags flags);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (MPSCnnBinaryConvolutionNode), Name = "MPSCNNBinaryFullyConnectedNode")]
+	interface MPSCnnBinaryFullyConnectedNode
+	{
+		[Static]
+		[Export ("nodeWithSource:weights:scaleValue:type:flags:")]
+		MPSCnnBinaryFullyConnectedNode FromSource (MPSNnImageNode sourceNode, IMPSCnnConvolutionDataSource weights, float scaleValue,MPSCnnBinaryConvolutionType type,MPSCnnBinaryConvolutionFlags flags);
+
+		[Export ("initWithSource:weights:scaleValue:type:flags:")]
+		IntPtr Constructor (MPSNnImageNode sourceNode, IMPSCnnConvolutionDataSource weights, float scaleValue,MPSCnnBinaryConvolutionType type,MPSCnnBinaryConvolutionFlags flags);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (MPSCnnConvolutionNode), Name = "MPSCNNConvolutionTransposeNode")]
+	interface MPSCnnConvolutionTransposeNode
+	{
+		[Static]
+		[Export ("nodeWithSource:convolutionState:weights:")]
+		MPSCnnConvolutionTransposeNode FromSource (MPSNnImageNode sourceNode, [NullAllowed]MPSCnnConvolutionStateNode convolutionState, IMPSCnnConvolutionDataSource weights);
```

Comment:
Methods should have verbs: `CreateFromSource`.

Reply:
Same